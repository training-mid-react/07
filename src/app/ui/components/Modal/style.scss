.modal {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.663);
    display: flex;
    align-items: center;
    justify-content: center;
    z-index: 10;
    &__container {
        background-color: #fff;
        padding: 50px;
        border-radius: 10px;
        box-shadow: 0 0 10px rgba(178, 178, 178, 0.932);
        width: 350px;
        text-align: center;
    }
    &__button{
        border: none;
        padding: 10px 20px;
        border-radius: 10px;
        background-color: #5679f7;
        color: #fff;
        cursor: pointer;
        &:hover{
            background-color: #fff;
            color: #5679f7;
        }
    }
}