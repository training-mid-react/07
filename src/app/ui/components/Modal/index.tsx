import "./style.scss"

interface Props {
    title: string;
    subtitle: string;
    playAgain: () => void;
}

const Modal = ({ title, subtitle, playAgain }: Props) => {
    return (
        <article className="modal">
            <div className="modal__container">
                <h1>{title}</h1>
                <p>{subtitle}</p>
                <button className="modal__button" onClick={playAgain}>Volver a jugar</button>
            </div>
        </article>
    );
}

export default Modal;