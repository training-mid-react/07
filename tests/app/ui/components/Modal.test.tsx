import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { vi } from "vitest";
import Modal from "../../../../src/app/ui/components/Modal";

describe("Test de Modal", () => {
    const mockPlayAgain = vi.fn();

    test("Snapshot del Modal", () => {
        const { asFragment } = render(<Modal title="Título" subtitle="Subtítulo" playAgain={mockPlayAgain} />)
        expect(asFragment).toMatchSnapshot();
    })

    test("Renderiza el Modal correctamente", () => {
        const { container } = render(<Modal title="Título" subtitle="Subtítulo" playAgain={mockPlayAgain} />)
        expect(container.querySelectorAll(".modal").length).toBe(1);
    })

    test("Renderiza el Modal con el título correcto", () => {
        render(<Modal title="Título" subtitle="Subtítulo" playAgain={mockPlayAgain} />)

         expect(screen.getByText('Título')).toBeTruthy();
    })

    test("Renderiza el Modal con el subtítulo correcto", () => {
        render(<Modal title="Título" subtitle="Subtítulo" playAgain={mockPlayAgain} />)
        expect(screen.getByText('Subtítulo')).toBeTruthy();
    })

    test("Llama al método de click del Modal", () => {
        render(<Modal title="Título" subtitle="Subtítulo" playAgain={mockPlayAgain} />)
        const buttton = screen.getByTestId('modal_button');
        fireEvent.click(buttton);
        expect(mockPlayAgain).toHaveBeenCalledTimes(1);
    })
})