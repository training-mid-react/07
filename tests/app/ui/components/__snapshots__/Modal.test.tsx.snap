// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test de Modal > Snapshot del Modal 1`] = `[Function]`;
